cranelift/frontend/src/frontend.rs

```diff
@@ -714,14 +713,14 @@ impl<'a> FunctionBuilder<'a> {
 
     /// Returns `true` if and only if no instructions have been added since the last call to
     /// `switch_to_block`.
-    pub fn is_pristine(&self) -> bool {
-        self.func_ctx.blocks[self.position.unwrap()].pristine
+    fn is_pristine(&self, block: Block) -> bool {
```

Comment:
Saltwater uses this function. (and so does it fork crabwater that aims to revive saltwater).

Reply:
I gather you mean https://github.com/nulla-git/crabwater/? I didn't know that existed.

Reply:
Yes

Reply:
After discussion with Chris, I'm going to merge this despite that breakage. For anyone who's using this API, I'd be happy to help with migrating to something else after a discussion about what they actually need from it.

Reply:
More specifically, to give some reasoning for this:
- We don't guarantee stability of Cranelift APIs between releases (each release is a semver break, and we haven't declared any sort of long-term stability guarantee generally);
- Thus, it's *technically* allowed to alter APIs arbitrarily;
- All the same, we want to minimize churn, as it's definitely in the project's interest to bring users along to new versions with minimal work. So we should have a reason to remove or alter APIs;
- In this case, @jameysharp 's reasoning (with which I think I agree) is that the function just seems a bit redundant or unnecessary: it lets the user query whether they have already added instructions, but it seems better either for the user to know that directly, or (possibly) to get that via more general IR introspection.

So on balance, I think this API cleanup makes sense, but we also don't want to cause arbitrary and unfixable breakage so please do let us know (anyone) if there are compatibility concerns we can try to address some other way.

Reply:
(also, is no one else chuckling about the puns in all these names?!)